### DISCUSSION TITLE
How do I swap from GPU to CPU?

### DISCUSSION THREAD

**Discussion by LiamPerson** [oobabooga/text-generation-webui#3126] [Q&A]
During setup I chose to use my GPU but now I want to use the CPU. How do I swap from GPU to CPU?

**Comment by CataCluj** [comment]
I would also like an answer to this, please. Anybody?
The GPU I chose seems to be wrong; there are no easy uninstallation instructions, I worry about making a mess.
SOMEBODY should know, or it's a problem, eh?

**Reply by lufixSch** [reply]
What do you mean with "The GPU I chose seems to be wrong"?

**Comment by lufixSch** [comment]
For llama.cpp you can simply choose not to offload any layers to GPU. If that doesn't work, you can tick the "CPU" checkbox. Then it will use llama.cpp specifically build for CPU

**Comment by CataCluj** [comment]
I chose AMD or rocm. I have an iGPU/APU that I can allocate 16GB RAM to. It should work and be faster than CPU. However, AMD in their greatness make it impossible (at least to me) to use. So I can't use it.
It installed rocm-based-torch. I can choose to offload or not until I'm blue, it won't work.
There should be an easier, documented way.

**Comment by araleza** [comment]
To uninstall, you can just delete the text-generation-webiu directory.  Because it uses a virtual environment, that will delete the application and all of its dependencies too. Again, because of the virtual environment, this deletion shouldn't affect anything else you have installed.

Do remember to get any models you've downloaded out of the models subdirectory first, if you want to keep them. Same for chats and character cards etc. that you've downloaded too.

**Reply by CataCluj** [reply]
Thank you @araleza.
I'm not sure what "webiu" means but I suppose you mean the whole directory, after I save my models elsewhere.
I always worry that even with venvs, something keeps track of them and that would be a loose end. I'm sure I don't understand all intricacies; maybe I'll learn in time.
